src/cgame/cg_gameinfo.cpp

```diff
@@ -31,6 +31,174 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 #include "common/FileSystem.h"
 #include "cg_local.h"
 
+//
+// arena info
+//
+static int  cg_numArenas;
+static char *cg_arenaInfos[ MAX_ARENAS ];
+
+/*
+===============
+CG_ParseInfos
+===============
+*/
+int CG_ParseInfos( const char *buf, int max, char *infos[] )
+{
+	char *token;
+	int  count = 0;
+	char key[ MAX_TOKEN_CHARS ];
+	char info[ MAX_INFO_STRING ];
+
+	auto infoPostfixLen = strlen( "\\num\\" ) + strlen( va( "%d", MAX_ARENAS ) );
+
+	while ( 1 )
+	{
+		token = COM_Parse( &buf );
+
+		if ( !token[ 0 ] )
+		{
+			break;
+		}
+
+		if ( strcmp( token, "{" ) )
+		{
+			Log::Notice( "Missing { in info file\n" );
+			break;
+		}
+
+		if ( count == max )
+		{
+			Log::Notice( "Max infos exceeded\n" );
+			break;
+		}
+
+		info[ 0 ] = '\0';
+
+		while ( 1 )
+		{
+			token = COM_ParseExt( &buf, true );
+
+			if ( !token[ 0 ] )
+			{
+				Log::Notice( "Unexpected end of info file\n" );
+				break;
+			}
+
+			if ( !strcmp( token, "}" ) )
+			{
+				break;
+			}
+
+			Q_strncpyz( key, token, sizeof( key ) );
+
+			token = COM_ParseExt( &buf, false );
+
+			if ( !token[ 0 ] )
+			{
+				strcpy( token, "<NULL>" );
+			}
+
+			Info_SetValueForKey( info, key, token, false );
+		}
+
+		//NOTE: extra space for arena number
+		infos[ count ] = (char*) BG_Alloc( strlen( info ) + infoPostfixLen + 1 );
+
+		if ( infos[ count ] )
+		{
+			strcpy( infos[ count ], info );
+			count++;
+		}
+	}
+
+	return count;
+}
+
+/*
+===============
+CG_LoadArenasFromFile
+TODO: consider deleting, unless we want to display 'longname' somewhere
+===============
+*/
+void CG_LoadArenasFromFile( char *filename )
```

Comment:
Do we want to enforce only one arena per file?

Currently it looks like we can have many arenas in a file. I don't think this is what we want or what we do in practice.